java/src/main/java/com/google/api/services/samples/youtube/cmdline/live/ListLiveChatMessages.java

```diff
@@ -0,0 +1,272 @@
+     * @param nextPageToken The page token from the previous request, if any.
+     * @param delayMs The delay in milliseconds before making the request.
+     */
+    private static void listChatMessages(
+        final String liveChatId,
+        final String nextPageToken,
+        long delayMs) {
+        System.out.println(
+            String.format("Getting chat messages in %1$.3f seconds...", delayMs * 0.001));
+        pollTimer = new Timer();
+        pollTimer.schedule(
+            new TimerTask() {
+                @Override
+                public void run() {
+                    try {
+                        // Get chat messages from YouTube
+                        LiveChatMessageListResponse response = youtube
+                            .liveChatMessages()
+                            .list(liveChatId, "snippet, authorDetails")
+                            .setPageToken(nextPageToken)
+                            .setFields(LIVE_CHAT_FIELDS)
+                            .execute();
+
+                        // Display messages and super chat details
+                        List<LiveChatMessage> messages = response.getItems();
+                        for (int i = 0; i < messages.size(); i++) {
+                            LiveChatMessage message = messages.get(i);
+                            LiveChatMessageSnippet snippet = message.getSnippet();
+                            System.out.println(buildOutput(
+                                snippet.getDisplayMessage(),
+                                message.getAuthorDetails(),
+                                snippet.getSuperChatDetails()));
+                        }
+
+                        // Request the next page of messages
+                        listChatMessages(
+                            liveChatId,
+                            response.getNextPageToken(),
+                            response.getPollingIntervalMillis());
+                    } catch (Throwable t) {
+                        System.err.println("Throwable: " + t.getMessage());
+                        t.printStackTrace();
+                    }
+                }
+            }, delayMs);
+    }
+
+    /**
+     * Lists live chat messages, polling at the client supplied interval. This method is not
+     * recommended because it will consume more API usage, but it may be necessary in some
+     * applications that require lower latency. Page tokens do not work when polling faster than the
+     * server supplied interval, so we need to keep track of the publish time to avoid duplicate
+     * message output. Message ids will not work for tracking the last received message because
+     * messages may be removed from chat.
+     *
+     * @param liveChatId The live chat id to list messages from.
+     * @param periodMs The fixed interval to poll messages.
+     * @param minPublishTime The minimum message time to output.
+     */
+    private static void listChatMessagesFixedPeriod(
```

Comment:
It was my understanding that the server rejects requests to poll faster than the polling interval, is that not the case?